homeassistant/components/homekit_controller/__init__.py

```diff
@@ -227,6 +228,16 @@ def get_characteristic_types(self):
     hass.data[CONTROLLER] = aiohomekit.Controller()
     hass.data[KNOWN_DEVICES] = {}
 
+    dothomekit_dir = os.path.join(hass.config.path(), ".homekit")
```

Comment:
Pass the custom path to `hass.config.path`. It will do the joining.

https://github.com/home-assistant/home-assistant/blob/b2d7bc40dc8c054430c6b8e0111be07060db5878/homeassistant/core.py#L1307-L1314